skimage/filters/ridges.py

```diff
@@ -197,26 +197,26 @@ def meijering(image, sigmas=range(1, 10, 2), alpha=None,
     for i, sigma in enumerate(sigmas):
 
         # Calculate (sorted) eigenvalues
-        eigenvalues = compute_hessian_eigenvalues(image, sigma, sorting='val')
+        eigenvalues = compute_hessian_eigenvalues(image, sigma, sorting='abs')
 
         if ndim > 1:
 
             # Set coefficients for scaling eigenvalues
             coefficients = [alpha] * ndim
             coefficients[0] = 1
 
-            # Compute auxiliary variables l_i = e_i + sum_{j!=i} alpha * e_j
+            # Compute normalized eigenvalues l_i = e_i + sum_{j!=i} alpha * e_j
             auxiliary = [np.sum([eigenvalues[i] * np.roll(coefficients, j)[i]
                          for j in range(ndim)], axis=0) for i in range(ndim)]
 
-            # Compute maximum over auxiliary variables
-            auxiliary = np.max(auxiliary, axis=0)
+            # Get maximum eigenvalues by magnitude
+            auxiliary = auxiliary[1]
 
-            # Rescale image intensity
-            filtered = np.abs(auxiliary) / np.abs(np.max(auxiliary))
+            # Rescale image intensity and avoid ZeroDivisionError
+            filtered = _divide_nonzero(auxiliary, np.min(auxiliary))
 
             # Remove background
-            filtered = np.where(filtered > 0, filtered, 0)
+            filtered = np.where(auxiliary < 0, filtered, 0)
```

Comment:
did you mean to flip this sign?

Reply:
Jep, the sign is reversed here, see screenshot above. (Roughly speaking, the opposite sign before this PR was "canceled out" by the reversed `if not black_ridges` at the top of the function. So there were in a way two sign changes in the old implementation, hence the results look similar and small deviations arise from the different eigenvalue sorting and rescaling.)